<launch>
  <arg name="joystick" 		default="true"/>
  <arg name="controller" 	default="true"/>
  <arg name="StrArduino" 		default="true"/>
  <arg name="DrvArduino" 		default="true"/>

  <group if="$(arg joystick)">
    <node name="joystick" pkg="joy" type="joy_node" required="true">
      <param name="dev" value="/dev/input/js2"/>
    </node>
  </group>

  <group if="$(arg controller)">
    <rosparam command="load" file="$(find joy_control)/config/params.yaml"/>
    <node name="controller4" pkg="joy_control" type="controller4" output="screen" required="true"/>
  </group>

  <group if="$(arg StrArduino)">
    <node pkg="rosserial_python" type="serial_node.py" name="StrArduino" required="true">
      <param name="port" value="/dev/ttyACM0"/>
      <param name="baud" value="57600"/>
    </node>
  </group>

  <group if="$(arg DrvArduino)">
    <node pkg="rosserial_python" type="serial_node.py" name="DrvArduino" required="true">
      <param name="port" value="/dev/ttyACM1"/>
      <param name="baud" value="115200"/>
    </node>
  </group>
</launch>